File: 2019/01/main.go
package main

import (
	"fmt"
	"io/ioutil"
	"strconv"
	"strings"
)

func main() {
	var totalFuel int
	data := readModules()
	for _, module := range data {
		totalFuel += fuel(int(module))
	}
	fmt.Println("Fuel needed for modules: ", totalFuel)

	var totalFuelPlusPlus int
	for _, module := range data {
		var f int = module
		for {
			f = fuel(f)
			if f <= 0 {
				break
			}
			totalFuelPlusPlus += f
		}
	}
	fmt.Println("Fuel needed for modules and fuel:", totalFuelPlusPlus)
}

func fuel(m int) int {
	return m/3 - 2
}

func readModules() []int {
	data, err := ioutil.ReadFile("input.txt")
	if err != nil {
		panic(err)
	}
	dataStrs := strings.Split(string(data), "\n")
	modules := make([]int, len(dataStrs)-1)
	for i, m := range dataStrs {
		if m != "" {
			modules[i], _ = strconv.Atoi(m)
		}
	}
	return modules
}

File: 2019/01/main.py
if __name__ == '__main__':
    with open('input.txt', 'r') as f:
        mods = f.readlines()
    
    mods = [int(m.replace('\n', '')) for m in mods]
    
    fuel = []
    for m in mods:
        f = m
        while f // 3 - 2 > 0:
            f = f // 3 - 2
            fuel.append(f)

    mods = sum([m // 3 - 2 for m in mods])
    print(f'Fuel needed for modules: {mods}')
    print(f'Fuel needed for modules and fuel: {sum(fuel)}')